src/view/com/composer/select-language/SelectPostLanguagesDialog.tsx

```diff
@@ -0,0 +1,418 @@
+    function mapCode2List(code2List: string[]) {
+      return code2List.map(code2 => LANGUAGES_MAP_CODE2[code2]).filter(Boolean)
+    }
+
+    // NOTE(@elijaharita): Get recent language codes and map them to language
+    // objects. Both the user account's saved language history and the current
+    // checked languages are displayed here.
+    const recentLanguagesCode2 =
+      Array.from(
+        new Set([...checkedLanguagesCode2, ...langPrefs.postLanguageHistory]),
+      ).slice(0, 5) || []
+    const recentLanguages = mapCode2List(recentLanguagesCode2)
+
+    // NOTE(@elijaharita): helper functions
+    const matchesSearch = (lang: Language) =>
+      lang.name.toLowerCase().includes(search.toLowerCase())
+    const isChecked = (lang: Language) =>
+      checkedLanguagesCode2.includes(lang.code2)
+    const isInRecents = (lang: Language) =>
+      recentLanguagesCode2.includes(lang.code2)
+
+    const checkedRecent = recentLanguages.filter(isChecked)
+
+    if (search) {
+      // NOTE(@elijaharita): if a search is active, we ALWAYS show checked
+      // items, as well as any items that match the search.
+      const uncheckedRecent = recentLanguages
+        .filter(lang => !isChecked(lang))
+        .filter(matchesSearch)
+      const unchecked = allowedLanguages.filter(lang => !isChecked(lang))
+      const all = unchecked
+        .filter(matchesSearch)
+        .filter(lang => !isInRecents(lang))
+
+      return {
+        all,
+        checkedRecent,
+        uncheckedRecent,
+      }
+    } else {
+      // NOTE(@elijaharita): if no search is active, we show everything.
+      const uncheckedRecent = recentLanguages.filter(lang => !isChecked(lang))
+      const all = allowedLanguages
+        .filter(lang => !recentLanguagesCode2.includes(lang.code2))
+        .filter(lang => !isInRecents(lang))
+
+      return {
+        all,
+        checkedRecent,
+        uncheckedRecent,
+      }
+    }
+  }, [
+    allowedLanguages,
+    search,
+    langPrefs.postLanguageHistory,
+    checkedLanguagesCode2,
+  ])
+
+  const listHeader = (
+    <View
+      style={[
+        {
+          paddingBottom: 8,
+          backgroundColor: t.atoms.bg.backgroundColor,
+        },
+        isNative && a.pt_2xl,
+      ]}>
+      <Text
+        nativeID="dialog-title"
+        style={[t.atoms.text, a.text_left, a.font_bold, a.text_xl, a.mb_sm]}>
+        <Trans>Choose Post Languages</Trans>
+      </Text>
+      <Text
+        nativeID="dialog-description"
+        style={[t.atoms.text_contrast_medium, a.text_left, a.text_md, a.mb_lg]}>
+        <Trans>Select up to 3 languages used in this post</Trans>
+      </Text>
+      <View style={[a.w_full, a.flex_row, a.align_stretch, a.gap_xs, a.pb_0]}>
+        <View style={[a.flex_1, a.relative]}>
+          <TextField.Root>
+            <TextField.Input
+              value={search}
+              onChangeText={setSearch}
+              placeholder="Search languages"
```

Comment:
```suggestion
              placeholder={_(msg`Search languages`)}
```
nit: Could this string be marked for translation please? 🙏